ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java

```diff
@@ -118,6 +121,8 @@
     this.jmxAdapter = new RaftServerJmxAdapter();
     this.leaderElectionMetrics = LeaderElectionMetrics.getLeaderElectionMetrics(this);
     this.raftServerMetrics = RaftServerMetrics.getRaftServerMetrics(this);
+
+    this.finishStart = new AtomicBoolean(false);
```

Comment:
Thanks @runzhiwang for discovering and providing this fix. This  is one of main reasons of several unit test failures.

The patch looks good to me.
Can we rename "finishStart" to something like startComplete or startDone? "finishStart" looks confusing tome. Can you also add a comment in the code why its important?

Reply:
@bshashikant Thanks for review. I have updated the patch.